presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -142,15 +152,22 @@ private synchronized void scheduleNextRequest()
             return;
         }
 
-        Request request = prepareGet()
+        Request request = setBinaryTransportHeadersIfNeeded(prepareGet())
                 .setUri(uriBuilderFrom(taskStatus.getSelf()).appendPath("status").build())
-                .setHeader(CONTENT_TYPE, JSON_UTF_8.toString())
                 .setHeader(PRESTO_CURRENT_STATE, taskStatus.getState().toString())
                 .setHeader(PRESTO_MAX_WAIT, refreshMaxWait.toString())
                 .build();
 
+        ResponseHandler responseHandler;
+        if (isBinaryTransportEnabled) {
+            responseHandler = createFullSmileResponseHandler((SmileCodec<TaskStatus>) taskStatusCodec);
+        }
+        else {
+            responseHandler = createAdaptingJsonResponseHandler(unwrapJsonCodec(taskStatusCodec));
+        }
+
```

Comment:
Can we pass a response handler based on the config? Instead of doing a if check.

Reply:
In theory we can, but not sure what the additional benefit will be.